File: exercise/php/day23/src/External/Stable/MagicStable.php
<?php

namespace External\Stable;

use External\Deer\Reindeer;

class MagicStable
{
    private Reindeer $dasher;
    private Reindeer $dancer;
    private Reindeer $prancer;
    private Reindeer $vixen;
    private Reindeer $comet;
    private Reindeer $cupid;
    private Reindeer $donner;
    private Reindeer $blitzen;
    private Reindeer $rudolph;

    public function __construct()
    {
        $this->dasher = new Reindeer("Dasher", 4, 10, true);
        $this->dancer = new Reindeer("Dancer", 2, 8);
        $this->prancer = new Reindeer("Prancer", 3, 9);
        $this->vixen = new Reindeer("Vixen", 3, 6);
        $this->comet = new Reindeer("Comet", 4, 9, true);
        $this->cupid = new Reindeer("Cupid", 4, 6);
        $this->donner = new Reindeer("Donner", 7, 6);
        $this->blitzen = new Reindeer("Blitzen", 8, 7);
        $this->rudolph = new Reindeer("Rudolph", 6, 3);
    }

    public function getAllReindeers(): array
    {
        return [
            $this->dasher,
            $this->dancer,
            $this->prancer,
            $this->vixen,
            $this->comet,
            $this->cupid,
            $this->donner,
            $this->blitzen,
            $this->rudolph
        ];
    }
}

File: exercise/C#/day23/ControlSystem.External/MagicStable.cs
namespace ControlSystem.External
{
    public class MagicStable
    {
        // Dasher is sick since yesterday
        private readonly Reindeer _dasher = new("Dasher", 4, 10, true);
        private readonly Reindeer _dancer = new("Dancer", 2, 8);
        private readonly Reindeer _prancer = new("Prancer", 3, 9);

        private readonly Reindeer _vixen = new("Vixen", 3, 6);

        // Comet is sick as well
        private readonly Reindeer _comet = new("Comet", 4, 9, true);
        private readonly Reindeer _cupid = new("Cupid", 4, 6);
        private readonly Reindeer _donner = new("Donner", 7, 6);
        private readonly Reindeer _blitzen = new("Blitzen", 8, 7);
        private readonly Reindeer _rudolph = new("Rudolph", 6, 3);

        public List<Reindeer> GetAllReindeers() =>
            [_dasher, _dancer, _prancer, _vixen, _comet, _cupid, _donner, _blitzen, _rudolph];
    }
}